### PR TITLE
Shadow segment creation

### PR DESCRIPTION
**Pull request by anju-c-das**
**Change log description**  
create shadow segment for event streams and related APIs to support functioning with shadow segment.

**Purpose of the change**  
(_e.g._, Fixes #666, Closes #1234)

**What the code does**  
- creates a new shadow segment for event streams. Initialize an "event number" attribute on the segment that can be checked and updated at append time.
- Binds the lifecycle of the shadow segment with parent segment.
- Enforce that appends to this segment are fixed size records.
- If the main segment is deleted, then shadow segment associated with it is also deleted.

### CODE REVIEW

File: segmentstore/contracts/src/main/java/io/pravega/segmentstore/contracts/SegmentType.java [hunk 1]
```diff
@@ -155,7 +159,7 @@ public boolean isSystem() {
     public boolean isCritical() {
         return (this.flags & ROLE_CRITICAL) == ROLE_CRITICAL;
     }
-
```
Comment (Bhupender-Y) [inline segmentstore/contracts/src/main/java/io/pravega/segmentstore/contracts/SegmentType.java:158]: NIT : extra space.

File: segmentstore/server/host/src/main/java/io/pravega/segmentstore/server/host/handler/PravegaRequestProcessor.java [hunk 2]
```diff
@@ -485,6 +490,18 @@ public void createSegment(CreateSegment createStreamSegment) {
                 });
     }
 
+    private CompletableFuture<Void> createIndexSegment(final String segmentName) {
+        Collection<AttributeUpdate> attributes = Arrays.asList(
+                new AttributeUpdate(CREATION_TIME, AttributeUpdateType.None, System.currentTimeMillis()),
+                new AttributeUpdate(ATTRIBUTE_SEGMENT_TYPE, AttributeUpdateType.None, SegmentType.INDEX_SEGMENT.getValue()),
+                new AttributeUpdate(EVENT_COUNT, AttributeUpdateType.Accumulate, 0L), // to store the number of events
+                new AttributeUpdate(ALLOWED_INDEX_SEG_EVENT_SIZE, AttributeUpdateType.Replace, 0L)
```
Comment (Bhupender-Y) [inline segmentstore/server/host/src/main/java/io/pravega/segmentstore/server/host/handler/PravegaRequestProcessor.java:498]: Will it be fixed for this child segment? If yes then AttributeUpdateType can be None.

**Review by Bhupender-Y — CHANGES_REQUESTED** [review]
Can we add test case to verify child segment existence ?

File: segmentstore/server/host/src/main/java/io/pravega/segmentstore/server/host/handler/PravegaRequestProcessor.java (hunk 2 above)
Comment (anju-c-das) [reply segmentstore/server/host/src/main/java/io/pravega/segmentstore/server/host/handler/PravegaRequestProcessor.java]: Addressed review comment

**Comment by anju-c-das** [comment]
closing this PR as new PR is opened for same.

**Pull request closed without merging**